tests/test_comments.py

```diff
@@ -280,3 +292,678 @@ def test_find_unread_does_not_include_deleted_comments(self):
+        self.guid.reload()
+
+        file_node = FileNode.resolve_class(self.provider, FileNode.FILE).get_or_create(destination['node'], self._format_path(destination['path'], file_id=self.file._id))
+        assert_equal(self.guid._id, file_node.get_guid()._id)
+        assert_equal(self.guid.referent.node._id, destination['node']._id)
+        file_comments = Comment.find(Q('root_target', 'eq', self.guid._id))
+        assert_equal(file_comments.count(), 1)
+
+    def test_comments_move_when_folder_moved_from_project_to_component(self):
+        source = {
+            'path': '/subfolder/',
+            'node': self.project,
+            'provider': self.provider
+        }
+        destination = {
+            'path': '/subfolder/',
+            'node': self.component,
+            'provider': self.provider
+        }
+        file_name = 'file.txt'
+        self._create_file_with_comment(node=source['node'], path='{}{}'.format(source['path'], file_name))
+        self.file.move_under(destination['node'].get_addon(self.provider).get_root())
+        payload = self._create_payload('move', self.user, source, destination, self.file._id)
+        update_file_guid_referent(self=None, node=destination['node'], event_type='addon_file_moved', payload=payload)
+        self.guid.reload()
+
+        file_node = FileNode.resolve_class(self.provider, FileNode.FILE).get_or_create(destination['node'], self._format_path('{}{}'.format(destination['path'], file_name), file_id=self.file._id))
+        assert_equal(self.guid._id, file_node.get_guid()._id)
+        file_comments = Comment.find(Q('root_target', 'eq', self.guid._id))
+        assert_equal(file_comments.count(), 1)
+
+    def test_comments_move_when_folder_moved_from_component_to_project(self):
+        source = {
+            'path': '/subfolder/',
+            'node': self.component,
+            'provider': self.provider
+        }
+        destination = {
+            'path': '/subfolder/',
+            'node': self.project,
+            'provider': self.provider
+        }
+        file_name = 'file.txt'
+        self._create_file_with_comment(node=source['node'], path='{}{}'.format(source['path'], file_name))
+        self.file.move_under(destination['node'].get_addon(self.provider).get_root())
+        payload = self._create_payload('move', self.user, source, destination, self.file._id)
+        update_file_guid_referent(self=None, node=destination['node'], event_type='addon_file_moved', payload=payload)
+        self.guid.reload()
+
+        file_node = FileNode.resolve_class(self.provider, FileNode.FILE).get_or_create(destination['node'], self._format_path('{}{}'.format(destination['path'], file_name), file_id=self.file._id))
+        assert_equal(self.guid._id, file_node.get_guid()._id)
+        file_comments = Comment.find(Q('root_target', 'eq', self.guid._id))
+        assert_equal(file_comments.count(), 1)
+
+    def test_comments_move_when_file_moved_to_osfstorage(self):
```

Comment:
The purpose of this test is to override the one in the mixin, which does not apply to osfstorage files.